internal/distributed/proxy/service_test.go

```diff
@@ -789,261 +790,365 @@ func Test_NewServer(t *testing.T) {
 	server.queryCoordClient = &MockQueryCoord{}
 	server.dataCoordClient = &MockDataCoord{}
 
+	var wg sync.WaitGroup
```

Comment:
Do not get how this wait group prevents the data race.  Without calling t.Parallel(), test cases run in sequence I think.